spec/features/search_spec.rb

```diff
@@ -17,20 +17,28 @@
 
     visit '/proposals'
     fill_in 'text', with: proposals.first.name
-    click_button 'Search'
+    click_button "search-button"
 
     expect(current_path).to eq('/proposals/query')
     expect(page).to have_content(proposals.first.public_id)
     expect(page).not_to have_content(proposals.last.name)
   end
 
-  it "populates the search box on the results page" do
+  it "populates the search box on the results page", :js do
     visit '/proposals'
     fill_in 'text', with: 'foo'
-    click_button 'Search'
+    click_button "search-button"
 
     expect(current_path).to eq('/proposals/query')
     field = find_field('text')
     expect(field.value).to eq('foo')
   end
+
+  it "does not show search UI for user without client_slug" do
+    no_client_user = create(:user)
+    login_as(no_client_user)
+
+    visit proposals_path
+    expect(page).not_to have_button("Search")
+  end
```

Comment:
feature test added in fd29c27. It was back-ported from #899 which has additional tests that address advanced search.